src/PrestaShopBundle/Translation/Provider/TranslationsCatalogueProvider.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShopBundle\Translation\Provider\Factory\ProviderFactory;
+use PrestaShopBundle\Translation\Provider\Type\TypeInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+
+/**
+ * Retrieves combined and formatted catalogues depending on the provider type.
+ */
+class TranslationsCatalogueProvider
+{
+    /**
+     * @var ProviderFactory
+     */
+    private $providerFactory;
+
+    public function __construct(ProviderFactory $providerFactory)
+    {
+        $this->providerFactory = $providerFactory;
+    }
+
+    /**
+     * @param TypeInterface $providerType
+     * @param string $locale
+     * @param string $domain
+     * @param array $search
+     *
+     * @return array
+     *
+     * @throws \PrestaShopBundle\Exception\NotImplementedException
+     */
+    public function getDomainCatalogue(
+        TypeInterface $providerType,
+        string $locale,
+        string $domain,
+        array $search = []
```

Comment:
Is it normal to have the search here? Should it be limited to the locale / domain and if needed having a search method?

Reply:
on the interface, if you search a string and then click on another domain, the search value remains until you delete it
So we can ask for a catalogue domain and search in it

We can have 2 methods : one getting the whole catalogue and another one which filter
But this method can only be used for `getDomainCatalogue`
In `getCatalogue` we add extra informations to the "catalogue" depending on the search result